qa-tests-backend/src/test/groovy/IntegrationsSplunkViolationsTest.groovy

```diff
@@ -119,9 +119,9 @@ class IntegrationsSplunkViolationsTest extends BaseSpecification {
         boolean hasNetworkViolation = false
         boolean hasProcessViolation = false
         def port = splunkDeployment.splunkPortForward.getLocalPort()
-        for (int i = 0; i < 15; i++) {
-            log.info "Attempt ${i} to get violations from Splunk"
-            def searchId = SplunkUtil.createSearch(port, "| from datamodel Alerts.Alerts")
+        for (int i = 0; i < 20; i++) {
```

Comment:
I think you can use the `withRetry` helper function here instead of a for loop.

Reply:
Using `withRetry` here would loose me the early continue in case we get no results back, correct?
Given that this is happening frequently (usually we get results after 4-5 tries), I lean towards keeping both loops. 
WDYT?

Reply:
🤔 I think you can achieve a similar behavior just something like:
```groovy
withRetry(20, 15) {
    response = getSearchResults()
    assert response != null
    // Check for violations
    assert violations
}
```
If the response is `null` we early continue by failing the assertion.